src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/ValidateArgumentsOfPublicMethodsTests.cs

```diff
@@ -6715,5 +6715,55 @@ public void M(object instance)
 }",
             }.RunAsync();
         }
+
+        [Fact, WorkItem(3634, "https://github.com/dotnet/roslyn-analyzers/issues/3634")]
+        public async Task NullConditionalAssignmentOperator_NullableEnableContext_NoDiagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+#nullable enable
+using System.Collections.Generic;
+
+public class C
+{
+    public void ParameterTest(Dictionary<string, string>? dict = null)
+    {
+        dict ??= new Dictionary<string, string>();
```

Comment:
Note: the linked issue doesn't assign `dict` here. It assigns a variable that is unclear from the issue where it's. Hence, the warning in issue is correct.

Reply:
The issue described by the person is unclear, that's why in my reply I am saying "assuming you meant...". If that's not the case, then the issue makes sense because the parameter isn't checked.